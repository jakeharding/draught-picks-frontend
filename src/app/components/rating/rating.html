<!-- Generated template for the RatingComponent component -->
<ion-row>
  <div (click)="updateRating(1)" [ngClass]="{'color-mug': rating?.rating >= 1, 'no-color-mug': rating?.rating < 1, 'large': size == 'large'}"></div>
  <div (click)="updateRating(2)" [ngClass]="{'color-mug': rating?.rating >= 2, 'no-color-mug': rating?.rating < 2, 'large': size == 'large'}"></div>
  <div (click)="updateRating(3)" [ngClass]="{'color-mug': rating?.rating >= 3, 'no-color-mug': rating?.rating < 3, 'large': size == 'large'}"></div>
  <div (click)="updateRating(4)" [ngClass]="{'color-mug': rating?.rating >= 4, 'no-color-mug': rating?.rating < 4, 'large': size == 'large'}"></div>
  <div (click)="updateRating(5)" [ngClass]="{'color-mug': rating?.rating >= 5, 'no-color-mug': rating?.rating < 5, 'large': size == 'large'}"></div>
</ion-row>
