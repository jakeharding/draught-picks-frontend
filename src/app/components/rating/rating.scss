:host {
  display: flex;
  flex-direction: row;

  .color-mug{
    background-image: url("../../../assets/imgs/rating/mug_color.png");
    background-size: cover;
    height: 30px;
    width: 30px;
    margin-right: 5px;
  }

  .no-color-mug {
    background-image: url("../../../assets/imgs/rating/mug.png");
    background-size: cover;
    height: 30px;
    width: 30px;
    margin-right: 5px;
  }

  div {
    flex: 1;
  }

  .large {
    height: 55px;
    width: 55px;
    margin-right: 10px;
  }

  @media (max-width: 576px) {
    .large {
      height: 50px;
      width: 50px;
      margin-right: 0px;
    }
  }
}
