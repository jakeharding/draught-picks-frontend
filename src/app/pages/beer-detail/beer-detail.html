<!--
  Generated template for the BeerDetailPage page.

  See http://ionicframework.com/docs/components/#navigation for more info on
  Ionic pages and navigation.
-->
<ion-header>
  <draught-header></draught-header>
</ion-header>


<ion-content padding>
  <ion-row>
    <ion-col size="12" sizeSm="10" offsetSm="1" sizeMd="8" offsetMd="2" sizeLg="6" offsetLg="3" sizeXl="4" offsetXl="4">
      <ion-row>
        <ion-col text-center>
          <h1>{{ beer.name }}</h1>
        </ion-col>
      </ion-row>

      <ion-row>
        <ion-col size="4" class="ion-text-center">
          <span class="sub-heading">
            ABV
          </span>
          <span class="sub-heading">
            <span *ngIf="beer.abv">{{ beer.abv }}%</span>
            <span *ngIf="!beer.abv">N/A</span>
          </span>
        </ion-col>

        <ion-col size="4" class="ion-text-center">
          <span class="sub-heading">
            IBU
          </span>
          <span class="sub-heading">
            <span *ngIf="beer.ibu"> {{ beer.ibu }}</span>
            <span *ngIf="!beer.ibu"> N/A</span>
          </span>
        </ion-col>

        <ion-col size="4" class="ion-text-center">
          <span class="sub-heading">
            Match
          </span>
          <span class="sub-heading">
            <span *ngIf="beer.recommended?.percent_match">{{beer.recommended?.percent_match}}%</span>
            <span *ngIf="!beer.recommended?.percent_match">N/A</span>
          </span>
        </ion-col>
      </ion-row>

      <ion-row>
        <ion-col sizeSm="12" offsetSm="0" sizeMd="9" offsetMd="2" class="ion-text-center">
          <ion-button (click)="createRecent()">I'm drinking this now!</ion-button>
        </ion-col>
      </ion-row>

      <ion-row>
        <ion-col sizeSm="12" offsetSm="0" sizeMd="9" offsetMd="2">
          <p *ngIf="hasRating">Your rating:</p>
          <p *ngIf="!hasRating">You haven't rated this beer yet. Tap to rate!</p>
        </ion-col>
      </ion-row>

      <ion-row>
        <ion-col sizeSm="12" offsetSm="2" sizeMd="9" offsetMd="2">
          <rating [rating]="beerRating" [size]="'large'" (setRating)="setRating($event)"></rating>
        </ion-col>
      </ion-row>

      <ion-row>
        <ion-col sizeSm="12" offsetSm="0" sizeMd="9" offsetMd="2">
          <p>Description:</p>
          <br>
          <p>{{ beer?.description }}</p>
        </ion-col>
      </ion-row>

      <ion-row>
        <ion-col sizeSm="12" offsetSm="0" sizeMd="9" offsetMd="2">
          <ion-item>
            <ion-label position="floating">
              Tell us what you like:
            </ion-label>
            <ion-textarea rows="5" [(ngModel)]="beerRating && beerRating.description"></ion-textarea>
          </ion-item>
        </ion-col>
      </ion-row>

      <ion-row class="ion-text-center">
        <ion-col sizeSm="12" offsetSm="0" sizeMd="9" offsetMd="2">
          <ion-button (click)="saveRatingDescription()">Submit</ion-button>
        </ion-col>
      </ion-row>

    </ion-col>
  </ion-row>

</ion-content>
