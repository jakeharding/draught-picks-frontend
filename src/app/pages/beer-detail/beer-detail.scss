:host {

  .sub-heading {
    display: block;
    font-size: 1.5em;
    font-weight: 500;
  }

  p {
    margin: 0px 0px;
  }

  ion-item {
    padding-left: 0px!important;
  }

}
