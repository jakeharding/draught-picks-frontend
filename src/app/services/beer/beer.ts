import { HttpClient } from '@angular/common/http';
import { Injectable } from '@angular/core';
import { environment as Env } from '../../../environments/environment';
import Beer from '../../models/Beer';
import PageResponse from '../../models/PageResponse';
import { LIMIT } from '../../directives/infinite-scroller/infinite-scroller';
import { EMPTY, Observable } from 'rxjs';
import { map } from 'rxjs/operators';
import { AutoCompleteService } from 'ionic4-auto-complete';
/*
  Generated class for the BeerProvider provider.

  See https://angular.io/guide/dependency-injection for more info on providers
  and Angular DI.
*/
@Injectable({
  providedIn: 'root',
})
export class BeerProvider implements AutoCompleteService {
  url: string;
  recentsUrl: string;
  recommendedUrl: string;
  labelAttribute = 'name'; // Attribute needed for the AutoCompleteService
  constructor(public http: HttpClient) {
    this.url = `${Env.REST_API_ROOT}/beers`;
    this.recentsUrl = `${Env.REST_API_ROOT}/recent-beers`;
    this.recommendedUrl = `${Env.REST_API_ROOT}/recommended-beers`;
  }

  /**
   * search function
   * Parameters: Observable of type Beer list
   * Sets the limit and offset for the infinite scrolling
   */
  search(params: any): Observable<Beer[]> {
    if (!params.limit) {
      params.limit = LIMIT;
    }
    if (!params.offset) {
      params.offset = 0;
    }
    return this.http.get(this.url, {params}).pipe(map(({results}: PageResponse<Beer>) => results));
  }

  /**
   * Method called in auto complete search bar.
   * @param beerName - Name of beer
   * @returns - List of beers
   */
  getResults(beerName: string) {
    if (beerName) {
      return this.search({search: beerName});
    }
    return EMPTY;
  }

  /**
   * recents function
   * Parameters: Observable of type Beer list
   * Sets the limit and offset for the infinite scrolling
   */
  recents(params: any): Observable<Beer[]> {
    if (!params.limit) {
      params.limit = LIMIT;
    }
    if (!params.offset) {
      params.offset = 0;
    }

    return this.http.get(this.recentsUrl, {params}).pipe(map(({results}: PageResponse<Beer>) => results));
  }

  /**
   * recommended function
   * Parameters: query params for pagination
   * Retrieves a page of recommended beers for an authenticated user.
   */
  recommended(params: any): Observable<Beer[]> {
    if (!params.limit) {
      params.limit = LIMIT;
    }
    if (!params.offset) {
      params.offset = 0;
    }
    return this.http.get(this.recommendedUrl, {params})
      .pipe(map(({results}: PageResponse<Beer>) => results));
  }

  /**
   * createRecent function
   * Parameters: Beer
   * returns the recent beer list
   */
  createRecent(recent: Beer) {
    return this.http.post(this.recentsUrl, {beer: recent.uuid}).toPromise();
  }

  /**
   * Retrieve a information about a beer.
   * @param uuid - ID of beer
   * @returns - Promise a beer
   */
  retrieve(uuid: string) {
    return this.http.get(`${this.url}/${uuid}`).toPromise();
  }


}
