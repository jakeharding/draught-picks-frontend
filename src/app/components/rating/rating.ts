import { Component, EventEmitter, Input, Output } from '@angular/core';
import BeerRating from '../../models/BeerRating';
import { RatingProvider } from '../../services/rating/rating';
import { ToastController } from '@ionic/angular';

/**
 * Generated class for the RatingComponent component.
 *
 * See https://angular.io/api/core/Component for more info on Angular
 * Components.
 */
@Component({
  selector: 'rating',
  templateUrl: 'rating.html',
  styleUrls: ['./rating.scss']
})
export class RatingComponent {

  constructor(public ratingProvider: RatingProvider, public toastController: ToastController) {}

  private static readonly LARGE = 'large';

  @Input() rating: BeerRating = { beer: 'beerId'} as BeerRating;
  @Input() size = 'small';
  @Output() setRating = new EventEmitter<BeerRating>();

  /**
   * updateRating function
   * Parameters: rating
   * updates the rating of a beer
   */
  updateRating(rating: number) {
    if (this.size === RatingComponent.LARGE) {
      const success = async beerRating => {
        this.rating = beerRating; // Update the beerRating returned from REST API
        const toast = await this.toastController.create({
          message: 'Your rating is saved.',
          duration: 3000,
          position: 'top',
          cssClass: 'success-toast'
        });
        await toast.present();
        this.setRating.emit(beerRating); // Update parent component
      };

      if (this.rating && this.rating.uuid) {
        return this.ratingProvider.partialUpdate({rating, uuid: this.rating.uuid} as BeerRating).then(success);
      } else {
        return this.createRating(this.rating.beer, rating).then(success);
      }
    }
  }

  /**
   * createRating function
   * Parameters: rating
   * creates a new rating for a beer if none exist
   */
  createRating(beer: string, rating: number) {
    const newRating = {beer, rating} as BeerRating;
    return this.ratingProvider.create(newRating);
  }
}
