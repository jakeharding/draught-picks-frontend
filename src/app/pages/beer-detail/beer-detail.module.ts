import { NgModule } from '@angular/core';
import { IonicModule } from '@ionic/angular';
import { BeerDetailPage } from './beer-detail';
import { ComponentsModule } from '../../components/components.module';
import { RouterModule } from '@angular/router';

@NgModule({
  declarations: [
    BeerDetailPage,
  ],
  imports: [
    IonicModule,
    RouterModule.forChild([{ path: '/beer/:uuid', component: BeerDetailPage}]),
    ComponentsModule,
  ],
})
export class BeerDetailPageModule {}
