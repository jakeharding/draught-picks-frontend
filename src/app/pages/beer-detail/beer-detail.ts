import { Component } from '@angular/core';
import { NavController } from '@ionic/angular';
import { BeerProvider } from '../../services/beer/beer';
import Beer from '../../models/Beer';
import BeerRating from '../../models/BeerRating';
import { RatingProvider } from '../../services/rating/rating';
import { BasePage } from '../BasePage';
import { ActivatedRoute, Router } from '@angular/router';
import { ToastProvider } from '../../services/toast/toast';

/**
 * Generated class for the BeerDetailPage page.
 *
 * See https://ionicframework.com/docs/components/#navigation for more info on
 * Ionic pages and navigation.
 */

@Component({
  selector: 'page-beer-detail',
  templateUrl: 'beer-detail.html',
  styleUrls: ['./beer-detail.scss']
})
export class BeerDetailPage extends BasePage {

  beer: Beer;
  beerRating: BeerRating;

  /**
   * Constructor for BeerDetailPage class
   * Gets the Beer data and beerRating information
   */
  constructor(public navCtrl: NavController, public beerProvider: BeerProvider, public ratingProvider: RatingProvider,
              private route: ActivatedRoute, private router: Router, private toastProvider: ToastProvider) {
    super(`/beers/${route.snapshot.paramMap.get('uuid')}`);
    if (this.router.getCurrentNavigation().extras.state) {
      this.beer = this.router.getCurrentNavigation().extras.state.beer as Beer;
    } else {
      this.beer = { uuid: route.snapshot.paramMap.get('uuid') } as Beer;
    }

    this.beerRating = {
      description: '',
      // Used as flag to tell if this beer ahs been rated since 0 is falsy.
      rating: 0,
      beer: this.beer.uuid,
    } as BeerRating;

    this.beerProvider.retrieve(this.beer.uuid).then((beer: Beer) => {
      this.beer = beer;
      if (beer.rating.length > 0) {
        this.beerRating = beer.rating[0];
      }
    });

    if (this.beer.rating && this.beer.rating.length > 0) {
      this.beerRating = this.beer.rating[0];
    }
  }

  /**
   * Creates a recent beer entry in the database.
   */
  createRecent() {
    const success = async (recent) => {
      this.toastProvider.successToast('We saved a record of this you! Tell us what you think!');
    };
    const error = async (e) => {
      this.toastProvider.errorToast();
    };
    this.beerProvider.createRecent(this.beer).then(success, error);
  }

  /**
   * Saves the description of the beer for the user.
   */
  saveRatingDescription() {
    const success = (rating) => {
      this.beerRating = rating;
      this.toastProvider.successToast('Your description has been saved!');
    };
    const error = (e) => {
      this.toastProvider.errorToast('Having trouble saving your description.');
    };

    if (this.beerRating.rating) {
      return this.ratingProvider.partialUpdate(this.beerRating).then(success, error);
    } else {
      return this.ratingProvider.create(this.beerRating).then(success, error);
    }
  }

  /**
   * Setter for setting the rating on the component.
   * @param rating - User's rating
   */
  setRating(rating: BeerRating): void {
    this.beerRating = rating;
  }
}
