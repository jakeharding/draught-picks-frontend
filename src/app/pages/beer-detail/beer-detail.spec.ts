import { ComponentFixture, TestBed } from '@angular/core/testing';
import { BeerDetailPage } from './beer-detail';
import { CUSTOM_ELEMENTS_SCHEMA } from '@angular/core';
import { NavController } from '@ionic/angular';
import { RatingComponent } from '../../components/rating/rating';
import { BeerProvider } from '../../services/beer/beer';
import { RatingProvider } from '../../services/rating/rating';
import Beer from '../../models/Beer';
import BeerRating from '../../models/BeerRating';
import { ActivatedRoute, Router } from '@angular/router';
import { mockActivatedRoute, mockRouter, mockToast, mockToastProvider } from '../../../../setup-jest';
import { ToastProvider } from '../../services/toast/toast';

/**
 * beer-detail.spec.ts
 *
 * Created by jake
 * Created on 11/18/18
 *
 * Test the BeerDetailPage.
 */

describe('BeerDetailPage', () => {
  let fixture: ComponentFixture<BeerDetailPage>;
  let component: BeerDetailPage;

  const mockRating = { uuid: 'aRating'} as BeerRating;
  const mockBeer = { uuid: 'beerId', name: 'BEER', description: 'GOOD', ibu: null, rating: [mockRating] } as Beer;
  const mockBeerProvider = {
    retrieve: jest.fn(() => Promise.resolve(mockBeer)),
    createRecent: jest.fn()
  };

  const mockRatingProvider = {
    partialUpdate: jest.fn(() => Promise.resolve()),
    create: jest.fn(() => Promise.reject('BOOM'))
  };

  beforeEach(() => {
    TestBed.configureTestingModule({
      schemas: [CUSTOM_ELEMENTS_SCHEMA],
      declarations: [BeerDetailPage, RatingComponent],
      providers: [
        { provide: NavController, useValue: {}},
        { provide: BeerProvider, useValue: mockBeerProvider},
        { provide: RatingProvider, useValue: mockRatingProvider},
        { provide: ActivatedRoute, useValue: mockActivatedRoute},
        { provide: Router, useValue: mockRouter},
        { provide: ToastProvider, useValue: mockToastProvider}
      ]
    }).compileComponents();
  });

  describe('page init', () => {
    it('should initialize the BeerDetailPage using the extras state', () => {
      mockRouter.getCurrentNavigation.mockReturnValue({extras: {state: {beer: mockBeer}}});
      fixture = TestBed.createComponent(BeerDetailPage);
      component = fixture.debugElement.componentInstance;
      expect(component).toBeTruthy();
      expect(fixture).toMatchSnapshot();
      expect(component.beer).toBe(mockBeer);
      // Assert properties are set correctly for a beer with ratings
      expect(component.beerRating).toBeDefined();
      expect(mockBeerProvider.retrieve).toHaveBeenCalledTimes(1);
      expect(component.beerRating).toEqual(mockRating);
    });

    it('should initialize the BeerDetailPage with the uuid only', () => {
      mockRouter.getCurrentNavigation.mockReturnValue({extras: {}});
      const uuidOnly = {uuid: 'test uuid'};
      mockActivatedRoute.snapshot.paramMap.get.mockReturnValue(uuidOnly.uuid);
      fixture = TestBed.createComponent(BeerDetailPage);
      component = fixture.debugElement.componentInstance;
      expect(component).toBeTruthy();
      expect(fixture).toMatchSnapshot();
      expect(component.beer).toEqual(uuidOnly);
    });
  });

  describe('test methods', () => {
    beforeEach(() => {
      fixture = TestBed.createComponent(BeerDetailPage);
      component = fixture.debugElement.componentInstance;
    });
    test('createRecent to create error toast', async () => {
      mockBeerProvider.createRecent.mockReturnValue(Promise.reject('error'));
      await component.createRecent();
      expect(mockToastProvider.errorToast).toHaveBeenCalledWith();
      expect(mockBeerProvider.createRecent).toHaveBeenCalledWith(component.beer);
    });

    test('createRecent to create success toast', async () => {
      mockBeerProvider.createRecent.mockReturnValue(Promise.resolve('success'));
      await component.createRecent();
      expect(mockToastProvider.successToast).toHaveBeenCalledTimes(1);
      expect(mockToastProvider.successToast).toHaveBeenCalledWith('We saved a record of this you! Tell us what you think!');
      expect(mockBeerProvider.createRecent).toHaveBeenCalledWith(component.beer);
    });

    test('saveRatingDescription should call ratingProvider.partialUpdate and display a success toast', async () => {
      mockRatingProvider.partialUpdate.mockResolvedValue();
      await component.saveRatingDescription();
      expect(mockToastProvider.successToast).toHaveBeenCalledTimes(1);
      expect(mockRatingProvider.partialUpdate).toHaveBeenCalledWith(mockRating);
    });

    test('saveRatingDescription should display a error toast', async () => {
      await component.saveRatingDescription();
      expect(mockToastProvider.errorToast).toHaveBeenCalledTimes(1);
      expect(mockToastProvider.errorToast).toHaveBeenCalledWith('Having trouble saving your description.');
      expect(mockRatingProvider.create).toHaveBeenCalledWith(mockRating);
    });

    it('should set the rating', () => {
      const rating = {} as BeerRating;
      component.setRating(rating);
      expect(component.beerRating).toEqual(rating);
    });
  });
});
