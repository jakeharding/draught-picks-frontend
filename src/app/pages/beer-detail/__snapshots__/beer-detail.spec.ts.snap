// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BeerDetailPage page init should initialize the BeerDetailPage using the extras state 1`] = `
<page-beer-detail
  beer={[Function Object]}
  beerProvider={[Function Object]}
  beerRating={[Function Object]}
  navCtrl={[Function Object]}
  ratingProvider={[Function Object]}
  route={[Function Object]}
  router={[Function Object]}
  toastProvider={[Function Object]}
>
  <ion-header>
    <draught-header />
  </ion-header><ion-content
    class="ion-padding"
  >
    <ion-row>
      <ion-col
        offsetlg="3"
        offsetmd="2"
        offsetsm="1"
        offsetxl="4"
        size="12"
        sizelg="6"
        sizemd="8"
        sizesm="10"
        sizexl="4"
      >
        <ion-row>
          <ion-col
            class="ion-text-center"
          >
            <h1>
              
            </h1>
          </ion-col>
        </ion-row>
        <ion-row>
          <ion-col
            class="ion-text-center"
            size="4"
          >
            <span
              class="sub-heading"
            >
               ABV 
            </span>
            <span
              class="sub-heading"
            >
              
              
            </span>
          </ion-col>
          <ion-col
            class="ion-text-center"
            size="4"
          >
            <span
              class="sub-heading"
            >
               IBU 
            </span>
            <span
              class="sub-heading"
            >
              
              
            </span>
          </ion-col>
          <ion-col
            class="ion-text-center"
            size="4"
          >
            <span
              class="sub-heading"
            >
               Match 
            </span>
            <span
              class="sub-heading"
            >
              
              
            </span>
          </ion-col>
        </ion-row>
        <ion-row>
          <ion-col
            class="ion-text-center"
            offsetmd="2"
            offsetsm="0"
            sizemd="9"
            sizesm="12"
          >
            <ion-button>
              I'm drinking this now!
            </ion-button>
          </ion-col>
        </ion-row>
        <ion-row>
          <ion-col
            offsetmd="2"
            offsetsm="0"
            sizemd="9"
            sizesm="12"
          >
            
            
          </ion-col>
        </ion-row>
        <ion-row>
          <ion-col
            offsetmd="2"
            offsetsm="2"
            sizemd="9"
            sizesm="12"
          >
            <rating>
              <ion-row>
                <div />
                <div />
                <div />
                <div />
                <div />
              </ion-row>
            </rating>
          </ion-col>
        </ion-row>
        <ion-row>
          <ion-col
            offsetmd="2"
            offsetsm="0"
            sizemd="9"
            sizesm="12"
          >
            <p>
              Description:
            </p>
            <br />
            <p>
              
            </p>
          </ion-col>
        </ion-row>
        <ion-row>
          <ion-col
            offsetmd="2"
            offsetsm="0"
            sizemd="9"
            sizesm="12"
          >
            <ion-item>
              <ion-label
                position="floating"
              >
                 Tell us what you like: 
              </ion-label>
              <ion-textarea
                rows="5"
              />
            </ion-item>
          </ion-col>
        </ion-row>
        <ion-row
          class="ion-text-center"
        >
          <ion-col
            offsetmd="2"
            offsetsm="0"
            sizemd="9"
            sizesm="12"
          >
            <ion-button>
              Submit
            </ion-button>
          </ion-col>
        </ion-row>
      </ion-col>
    </ion-row>
  </ion-content>
</page-beer-detail>
`;

exports[`BeerDetailPage page init should initialize the BeerDetailPage with the uuid only 1`] = `
<page-beer-detail
  beer={[Function Object]}
  beerProvider={[Function Object]}
  beerRating={[Function Object]}
  navCtrl={[Function Object]}
  ratingProvider={[Function Object]}
  route={[Function Object]}
  router={[Function Object]}
  toastProvider={[Function Object]}
>
  <ion-header>
    <draught-header />
  </ion-header><ion-content
    class="ion-padding"
  >
    <ion-row>
      <ion-col
        offsetlg="3"
        offsetmd="2"
        offsetsm="1"
        offsetxl="4"
        size="12"
        sizelg="6"
        sizemd="8"
        sizesm="10"
        sizexl="4"
      >
        <ion-row>
          <ion-col
            class="ion-text-center"
          >
            <h1>
              
            </h1>
          </ion-col>
        </ion-row>
        <ion-row>
          <ion-col
            class="ion-text-center"
            size="4"
          >
            <span
              class="sub-heading"
            >
               ABV 
            </span>
            <span
              class="sub-heading"
            >
              
              
            </span>
          </ion-col>
          <ion-col
            class="ion-text-center"
            size="4"
          >
            <span
              class="sub-heading"
            >
               IBU 
            </span>
            <span
              class="sub-heading"
            >
              
              
            </span>
          </ion-col>
          <ion-col
            class="ion-text-center"
            size="4"
          >
            <span
              class="sub-heading"
            >
               Match 
            </span>
            <span
              class="sub-heading"
            >
              
              
            </span>
          </ion-col>
        </ion-row>
        <ion-row>
          <ion-col
            class="ion-text-center"
            offsetmd="2"
            offsetsm="0"
            sizemd="9"
            sizesm="12"
          >
            <ion-button>
              I'm drinking this now!
            </ion-button>
          </ion-col>
        </ion-row>
        <ion-row>
          <ion-col
            offsetmd="2"
            offsetsm="0"
            sizemd="9"
            sizesm="12"
          >
            
            
          </ion-col>
        </ion-row>
        <ion-row>
          <ion-col
            offsetmd="2"
            offsetsm="2"
            sizemd="9"
            sizesm="12"
          >
            <rating>
              <ion-row>
                <div />
                <div />
                <div />
                <div />
                <div />
              </ion-row>
            </rating>
          </ion-col>
        </ion-row>
        <ion-row>
          <ion-col
            offsetmd="2"
            offsetsm="0"
            sizemd="9"
            sizesm="12"
          >
            <p>
              Description:
            </p>
            <br />
            <p>
              
            </p>
          </ion-col>
        </ion-row>
        <ion-row>
          <ion-col
            offsetmd="2"
            offsetsm="0"
            sizemd="9"
            sizesm="12"
          >
            <ion-item>
              <ion-label
                position="floating"
              >
                 Tell us what you like: 
              </ion-label>
              <ion-textarea
                rows="5"
              />
            </ion-item>
          </ion-col>
        </ion-row>
        <ion-row
          class="ion-text-center"
        >
          <ion-col
            offsetmd="2"
            offsetsm="0"
            sizemd="9"
            sizesm="12"
          >
            <ion-button>
              Submit
            </ion-button>
          </ion-col>
        </ion-row>
      </ion-col>
    </ion-row>
  </ion-content>
</page-beer-detail>
`;
